### PR TITLE
some more implementions

### PR DESCRIPTION
**Pull request by lidaobing**

### CODE REVIEW

File: gtk/gtk.go [hunk 1]
```diff
@@ -5514,8 +5514,14 @@ func _go_call_sort_func(gsfi *C._gtk_sort_func_info) {
 	gsfi.ret = C.int(gots.sortFuncs[int(gsfi.columnNum)](&TreeModel{gsfi.model}, &a, &b))
 }
 
-// gtk_tree_sortable_set_default_sort_func
-// gtk_tree_sortable_has_default_sort_func
+func (ts *TreeSortable) SetDefaultSortFunc(fun SortFunc) {
+	ts.sortFuncs[TREE_SORTABLE_DEFAULT_SORT_COLUMN_ID] = fun
+	C._gtk_tree_sortable_set_default_sort_func(ts.GTreeSortable, pointer.Save(&ts))
```
Comment (mattn) [inline gtk/gtk.go]: How about to add new private struct to support user_data like below?

```
type sortFuncInfo struct {
	fun SortFunc
	user_data interface{}
}

type TreeSortable struct {
	GTreeSortable *C.GtkTreeSortable
	sortFuncs     map[int]sortFuncInfo
}
```

Comment (lidaobing) [reply gtk/gtk.go]: @mattn so we need change `SortFunc` from `type SortFunc func(m *TreeModel, a *TreeIter, b *TreeIter) int` to `type SortFunc func(m *TreeModel, a *TreeIter, b *TreeIter, data interface{}) int` ?

Comment (lidaobing) [reply gtk/gtk.go]: the python[1] and vala[2] binding both contain the user_data, maybe we also need keep it. 

[1] http://www.pygtk.org/pygtk2reference/class-gtktreesortable.html#method-gtktreesortable--set-sort-func
[2] https://valadoc.org/gtk+-3.0/Gtk.TreeSortable.set_default_sort_func.html

**Comment by mattn** [comment]
Great. Thank you.

**Pull request merged**